src/Factory/AwsFactory.php

```diff
@@ -16,18 +15,20 @@ class AwsFactory implements FactoryInterface
 {
     /**
      * @param ContainerInterface $container
-     * @param string             $requestedName
-     * @param array|null         $options
+     * @param string $requestedName
+     * @param array|null $options
      * @return AwsSdk
      */
     public function __invoke(ContainerInterface $container, $requestedName, array $options = null)
     {
         // Instantiate the AWS SDK for PHP
         $config = $container->get('Config');
         $config = isset($config['aws']) ? $config['aws'] : [];
+
+        $zend_framework_version = isset($config['aws_zf2']['zend_framework_version']) ? $config['aws_zf2']['zend_framework_version'] : '3.0';
         $config += [
             'ua_append' => [
-                'ZF2/' . Version::VERSION,
+                'ZF2/' . $zend_framework_version,
```

Comment:
@cjyclaire thank you for your notes!
The current implementation is as follows:
`
'ua_append' => [
                'ZF2/' . Version::VERSION,
                'ZFMOD/' . Module::VERSION,
 ]
`

Please note, we have 2 different version constants. The first one is Zend\Version which is taken from the 'zend-version' module and the second one is "module" version witch is hard-coded in Module.php. 

The only place the Zend Framework version (the first one) is currently used is to be added to the client user agent headers (ua_append) in AwsFactory.php. 

If we remove this header and leave it only with "ZFMOD" header, then I can remove all the additional code related to the added configuration parameter and the dependency on Zend\Version will be resolved.    

The other option is to pass the module version instead of the framework version. This is going to be backward compatible if somebody is parsing the "ua_append" header and is expecting to find "ZF2" parameter there.